src/ed25519.rs

```diff
@@ -431,15 +433,47 @@ impl<'d> Deserialize<'d> for Keypair {
             where
                 E: SerdeError,
             {
+                if bytes.len() != KEYPAIR_LENGTH {
+                    return Err(SerdeError::invalid_length(bytes.len(), &self));
+                }
+
                 let secret_key = SecretKey::from_bytes(&bytes[..SECRET_KEY_LENGTH]);
                 let public_key = PublicKey::from_bytes(&bytes[SECRET_KEY_LENGTH..]);
 
-                if secret_key.is_ok() && public_key.is_ok() {
-                    Ok(Keypair{ secret: secret_key.unwrap(), public: public_key.unwrap() })
+                if let (Ok(secret), Ok(public)) = (secret_key, public_key) {
+                    Ok(Keypair{ secret, public })
                 } else {
                     Err(SerdeError::invalid_length(bytes.len(), &self))
                 }
             }
+
+            fn visit_seq<A>(self, mut seq: A) -> Result<Keypair, A::Error>
+            where
+                A: SeqAccess<'d>
+            {
+                if let Some(len) = seq.size_hint() {
+                    if len != KEYPAIR_LENGTH {
+                        return Err(SerdeError::invalid_length(len, &self));
+                    }
+                }
+
+                // TODO: We could do this with `MaybeUninit` to avoid unnecessary initialization costs
+                let mut bytes: [u8; KEYPAIR_LENGTH] = [0u8; KEYPAIR_LENGTH];
```

Comment:
Are you okay with the use of `unsafe`, and the use of the relatively modern `MaybeUninit`? This could help to avoid having to initialize the 64 byte array to zero unnecessarily.

Reply:
The current (as of 1.44 stable) [docs](https://doc.rust-lang.org/core/mem/union.MaybeUninit.html) say "Notice that the rules around uninitialized integers are not finalized yet, but until they are, it is advisable to avoid them." so I think I'd be a little bit too uncomfortable with starting to use `MaybeUninit` at the moment.